src/plugin.rs

```diff
@@ -800,7 +800,7 @@ impl PluginParameters for DummyPluginParameters {}
 /// impl Default for ExamplePlugin {
 ///     fn default() -> ExamplePlugin {
 ///         // Will panic, don't do this. If needed, you can query
-///         // the host during initialization via Vst::new()
+///         // the host during initialization in `Plugin::new()`
 ///         let host: HostCallback = Default::default();
 ///         let version = host.vst_version();
```

Comment:
I'd suggest removing most of this example, since implementing `Default` for a `Plugin` is no longer needed and also not the recommended way of doing things.

Until we remove the `Default` implementation for `HostCallback` (if we decide to do that), the description could say something like
```rust
/// All methods in this struct will panic if the `HostCallback` was constructed using
/// `Default::default()` rather than being passed in via `Plugin::new`.
```
and then just have the two lines here and nothing else as the example.